doc/reference/reference_lua/datetime.rst

```diff
@@ -1,62 +1,955 @@
     ..  rst-class:: left-align-column-2
 
     ..  list-table::
-        :widths: 25 75
-        :header-rows: 1
+        :widths: 35 65
+
+        *   -   **Functions**
+            -
+
+        *   -   :ref:`datetime.new() <datetime-new>`
+            -   Create an object of the ``datetime`` type from a table of time units
+
+        *   -   :ref:`datetime.now() <datetime-now>`
+            -   Create an object of the ``datetime`` type with the current date and time
+
+        *   -   :ref:`datetime.is_datetime() <datetime-is_datetime>`
+            -   Check whether the specified value is a ``datetime`` object
+
+        *   -   :ref:`datetime.parse() <datetime-parse>`
+            -   Convert an input string with the date and time information into a ``datetime`` object
+
+        *   -   :ref:`datetime.interval.is_interval() <datetime-interval-is_interval>`
+            -   Check whether the specified value is an ``interval`` object
+
+        *   -   :ref:`datetime.interval.new() <datetime-interval-new>`
+            -   Create an object of the ``interval`` type from a table of time units
+
+        *   -   **Properties**
+            -
+
+        *   -   :ref:`datetime.TZ <datetime-tz>`
+            -   An array of timezone names and abbreviations
+
+        *   -   **Related objects**
+            -
+
+        *   -   :ref:`datetime_object <datetime_obj>`
+            -   A ``datetime`` object
+
+        *   -   :ref:`interval_object <interval_obj>`
+            -   An ``interval`` object
+
+
+
+.. module:: datetime
+
+..  _datetime-module-api-reference-functions:
+
+Functions
+~~~~~~~~~
+
+..  _datetime-new:
+
+..  function:: datetime.new( [{ units }] )
+
+    Create an object of the :ref:`datetime type <index-box_datetime>` from a table of time units.
+    See the :ref:`description of units <datetime-new-args>` and :ref:`examples <datetime-new-example>` below.
+
+    :param table units: Table of :ref:`time units <datetime-new-args>`.
+                                If an empty table or no arguments are passed, the ``datetime`` object with the default values corresponding to Unix Epoch is created: ``1970-01-01T00:00:00Z``.
+
+    :return: :ref:`datetime object <datetime_obj>`
+    :rtype: cdata
+
+    ..  _datetime-new-args:
+
+    **Possible input time units for datetime.new()**
+
+    ..  container:: table
+
+        ..  list-table::
+            :widths: 20 50 20 10
+            :header-rows: 1
+
+            *   -   Name
+                -   Description
+                -   Type
+                -   Default
+
+            *   -   nsec (usec, msec)
+                -   Fractional part of the last second. You can specify either nanoseconds (``nsec``), or microseconds (``usec``), or milliseconds (``msec``).
+                    Specifying two of these units simultaneously or all three ones lead to an error.
+                -   number
+                -   0
+
+            *   -   sec
+                -   Seconds. Value range: 0 - 60.
```

Comment:
Shouldn't we mention "leap second" here?

We have 0-59 minutes and 0-60 seconds. It can confuse newcomers.

Reply:
Thanks. Looks like this requires some investigation. Created a separate issue for this: https://github.com/tarantool/doc/issues/4409.